rockets/jsonrpc/requestProcessor.cpp

```diff
@@ -101,7 +113,10 @@ json RequestProcessor::_processValidBatchBlocking(const json& array,
                 responses.push_back(response);
         }
         else
-            responses.push_back(makeErrorResponse(invalidRequest));
+        {
+            const auto id = entry.count("id") ? entry["id"] : json();
+            responses.push_back(makeErrorResponse(invalidRequest, id));
```

Comment:
same question here? Is there a test that shows an actual invalid request with an id returned?